clients/js/package.json

```diff
@@ -65,5 +65,9 @@
     "ts-node": "^10.9.2",
     "typedoc": "^0.25.4",
     "typescript": "^4.8.3"
+  },
+  "peerDependencies": {
+    "ethers": "6.x",
```

Comment:
Hmm. Interesting that this works with v5.

Reply:
The idea is that if ethers v6 is installed it will use the same version.

I guess alternatively we could change the ethers dependency to `6.x` and remove the peerDependency.

If the users app uses ethers v5 and they install sapphire-paratime package their bundled dApp will still end up bundling both v5 & v6 which seems unavoidable at this point unless we completely remove ethers from sapphire-paratime...

This should be possible if we remove:

```
if (typeof upstream === 'string') {
    upstream = new JsonRpcProvider(upstream) as any;
  }
```

And then extract the needed types from ethers, and the utility functions `hexlify`, `isBytesLike`, `getBytes` etc.

Reply:
I did some reading and I'm still unsure of best practice for ensuring that if ethers v6 is installed we use whatever version the dapp is using

Reply:
Agree to stay with status quo for now.